horovod/common/controller.cc

```diff
@@ -632,30 +646,36 @@ ResponseList Controller::FuseResponses(std::deque<Response>& responses) {
     responses.pop_front();
     int64_t tensor_size = 0;
     DataType dtype;
-    if (response.response_type() == Response::ResponseType::ALLREDUCE || 
+    if (response.response_type() == Response::ResponseType::ALLREDUCE ||
         response.response_type() == Response::ResponseType::ADASUM) {
       // Attempt to add more responses to this fused response.
 
-      // found_tensor can be false for ranks that did Join.
-      bool found_tensor = tensor_queue_.GetTensorSizeAndType(response.tensor_names()[0], tensor_size, dtype);
```

Comment:
@kit1980 In some recent testing, I noticed that tensor fusion is currently not functional (i.e. no tensors are being fused). It appears the issue is that the `dtype` variable above is no longer being set to anything. Was the line calling `GetTensorSizeAndType` removed by mistake?

Reply:
Yes, my mistake. The result `found_tensor` in no longer needed because of the `joined` parameter, but the call is needed. 

I thought there is a test for fusion? I'll try to fix this for both join and non-join ASAP.

Reply:
There is a test indeed, `test_horovod_allreduce_async_fused` for PyTorch. But looks like it doesn't actually test the tensors are fused (and it probably can't via Python API?)

Reply:
Yeah, unfortunately that test assumes fusion buffering is occurring, and then tests to ensure that it produces correct results.  If fusion buffering is not working, then it will do nothing.  Once we move our build system to CMake, I want to introduce some C++-specific unit tests, of which this would be one of them, but for now it is difficult to test automatically.

Reply:
I've sent a PR fixing this: https://github.com/horovod/horovod/pull/1594